### PR TITLE
MC38472 remote storage

### PR DESCRIPTION
**Pull request by hguthrie**
This pull request (PR) adds new topics that explain the Remote storage module and AWS S3 storage adapter. It is prepared in such a way that it is easy to add new adapters to the list.

This PR add the following:

- a new section titled **Remote storage** to the Configuration guide TOC
- a new topic for the AWS S3 storage adapter
- a new topic for server-side image resizing
- a new topic for remote storage caching

Needs:

- [x] final technical approval
- [x] editorial approval

whatsnew
Add new [configuration instructions for the Remote storage module](https://devdocs.magento.com/guides/v2.4/config-guide/remote-storage/) and AWS S3 adapter to the _Configuration guide_.

### CODE REVIEW

**Review by adifucan — COMMENTED** [review]
Left some minor comments to extend documentation a bit.

**Comment by hguthrie** [comment]
Will also wrap up the review in the User guide: https://github.com/magento/merchdocs/pull/856

**Comment by hguthrie** [comment]
Corresponding Merchdocs PR: https://github.com/magento/merchdocs/pull/900

**Comment by hguthrie** [comment]
@shiftedreality Is the content all in? Do you have any further thing to add?
@jeff-matthews @bdenham Docs code freeze is Friday? So, if that is the case, then I need to edit and push this by Thursday. Can someone review?

**Comment by jeff-matthews** [comment]
@hguthrie, we don't technically have a docs code freeze. However, all docs for the 2.4.2 release should be merged by our release-to-marketing (RTM) deadline, which is January 12th.

**Comment by bdenham** [comment]
On my list.

File: src/guides/v2.4/config-guide/remote-storage/config-remote-storage.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+group: configuration-guide
+title: Configure remote storage
+functional_areas:
+  - Configuration
+  - System
+  - Setup
+---
+
+The Remote Storage module provides the option to store media files and schedule imports/exports in a persistent, remote storage container using a storage service, such as AWS S3 or Azure Blob Storage. By default, Magento stores media files in the same filesystem that contains the application. This is not efficient for complex, multi-server configurations, which can result in degraded performance when sharing resources. With this module, you can store media files in the `pub/media` directory and import/export files in the `var` directory in the remote object storage and take advantage of server-side image resizing.
```
Comment (bdenham) [inline src/guides/v2.4/config-guide/remote-storage/config-remote-storage.md:10]:
```suggestion
The Remote Storage module provides the option to store media files and schedule imports/exports in a persistent, remote storage container using a storage service, such as AWS S3 or Azure Blob Storage. By default, Magento stores media files in the same filesystem that contains the application. This is inefficient for complex, multi-server configurations, and can result in degraded performance when sharing resources. With the Remote Storage module, you can store media files in the `pub/media` directory and import/export files in the `var` directory of the remote object storage to take advantage of server-side image resizing.
```

File: src/guides/v2.4/config-guide/remote-storage/config-remote-storage.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+| `remote-storage-driver` | driver | Adapter name<br>Possible values:<br>**file**: Disables remote storage and uses the local filesystem<br>**aws-s3**: Use the [Amazon Simple Storage Service (Amazon S3)][AWS S3] | none |
+| `remote-storage-bucket` | bucket | Object storage or container name | none |
+| `remote-storage-prefix` | prefix | Optional prefix (location inside of object storage) | empty |
+| `remote-storage-region` | region | Region name | none |
+| `remote-storage-key` | access key | Optional access key | empty |
+| `remote-storage-secret` | secret key | Optional secret key | empty |
+
+### Storage adapters
+
+The default storage location is the local filesystem. A _storage adapter_ enables you to connect to a storage service and store your files anywhere. Magento supports configuring the following storage services:
```
Comment (bdenham) [inline src/guides/v2.4/config-guide/remote-storage/config-remote-storage.md:35]:
```suggestion
The default storage location is in the local filesystem. A _storage adapter_ enables you to connect to a storage service and store your files anywhere. Magento supports configuring the following storage services:
```

File: src/guides/v2.4/install-gde/install/cli/install-cli-install.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ The install command uses the following format:
 magento setup:install --<option>=<value> ... --<option>=<value>
 ```
 
-The following tables describe the installation option names and values. For example installation commands. see [Sample localhost installations](#install-cli-example).
+The following tables describe the installation option names and values. For example installation commands. See [Sample localhost installations](#install-cli-example).
```
Comment (bdenham) [inline src/guides/v2.4/install-gde/install/cli/install-cli-install.md:69]:
```suggestion
The following tables describe the installation option names and values, such as installation commands. See [Sample localhost installations](#install-cli-example).
```

**Review by bdenham — APPROVED** [review]
Approved pending suggested changes at your discretion.

**Comment by hguthrie** [comment]
running tests

**Pull request merged**